tests/test_convert_diann.py

```
@@ -213,9 +214,184 @@ def test_cli_convert_diann_label_free(self, diann_data_dir, on_tmpdir):
+        df = pd.read_csv("diann_design.tsv", sep="\t")
+        assert len(set(df["BioReplicate"].tolist())) == 3
+
+    def test_mtraq_per_channel_samples(self, diann_data_dir, on_tmpdir):
+        DiaNN().diann_convert(str(diann_data_dir / "mtraq_design.sdrf.tsv"))
+        df = pd.read_csv("diann_design.tsv", sep="\t")
+        assert len(set(df["Sample"].tolist())) == 3
+
+
+class TestDiannScanRangeValidation:
+    def test_inverted_scan_range_raises_error(self, diann_data_dir, on_tmpdir):
+        with pytest.raises(ValueError, match="[Ii]nverted|[Mm]in.*greater.*max"):
```

Ruff (0.15.6)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)